app/models/user/profile.rb

```diff
@@ -19,4 +19,9 @@ def testimonials_tab?
   def contributions_tab?
     user.reputation_tokens.where(category: %i[building authoring maintaining misc]).exists?
   end
+
+  memoize
+  def badges_tab?
+    user.acquired_badges.revealed.exists?
```

Comment:
On other locations, `.count.positive?` is used. I prefer `exists?` as it feels a bit more semantic (we're not really interested in the actual count).

The SQL is almost identical:

*.exists?*
```sql
SELECT 1 AS one FROM `mentor_testimonials` WHERE `mentor_testimonials`.`mentor_id` = 4 AND `mentor_testimonials`.`revealed` = TRUE AND `mentor_testimonials`.`deleted_at` IS NULL LIMIT 1;
```

*.count.positive?*
```sql
SELECT COUNT(*) FROM `mentor_testimonials` WHERE `mentor_testimonials`.`mentor_id` = 4 AND `mentor_testimonials`.`revealed` = TRUE AND `mentor_testimonials`.`deleted_at` IS NULL;
```

Checking their query plans reveals them to have an identical cost.

Reply:
`exists` is definitely better.